### ISSUE TITLE
False positive for UnusedPrivateMethod when method is called on clone

### ISSUE THREAD

**Issue by sveneld** [phpmd/phpmd#937]
- PHPMD version: 2.11.0
- PHP Version: 7.4.3
- Installation type: composer
- Operating System / Distribution & Version: Ubuntu 20.04.3 LTS

### Current Behavior
`<?php

class Test
{

    private $keywords = [];

    public function testClone()
    {
        $clone = clone $this;
        $clone->prepareKeywords();

        return $clone;
    }

    private function prepareKeywords()
    {
        $this->keywords = ['a', 'b'];
    }
}
`
phpmd: Avoid unused private methods such as 'prepareKeywords'.

### Expected Behavior
No error

### Steps To Reproduce:
Run phpmd inspection on provided code

**Comment by tvbeek** [comment]
@sveneld thanks for your report, this is an interesting case. If this function is only used after the clone I should personal use the `__ clone` function. See: https://www.php.net/clone
I'm not sure if we can check this correct, do you have the possibility to create a PR with a failing test? That can be a nice starting point.


ps. I have edited your post to have the code highlighting working.

**Comment by sveneld** [comment]
Ok, I will try to create PR with failing test

**Comment by sveneld** [comment]
@tvbeek https://github.com/phpmd/phpmd/pull/940 here it is

**Comment by cs278** [comment]
Basically the problem (using 2.12.0), by creating a new instance in a static method:

> Avoid unused private methods such as 'doThing'

```php
final class Test
{
    public static function create(): self
    {
        $obj = new self();
        $obj->doThing();

        return $obj;
    }

    private function doThing(): void
    {
    }
}
```

**Comment by AJenbo** [comment]
Duplicate of #252
